### PR TITLE
feat(deployment): add restify adapter

### PR DESCRIPTION
**Pull request by jenseng**
copy-pasta/tweaked from express adapter

testing notes:
☑️  standard adapter tests, plus some more stream ones
☑️  generated a restify-remix app via create-remix and manually verified

lmk if there's anything I should add :)

### CODE REVIEW

**Review by w3bdesign — APPROVED** [review]

File: packages/create-remix/templates/restify/remix.config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @type {import('@remix-run/dev/config').AppConfig}
+ */
+module.exports = {
+  appDirectory: "app",
+  browserBuildDirectory: "public/build",
```
Comment (MichaelDeBoey) [inline packages/create-remix/templates/restify/remix.config.js:6]:
```suggestion
  assetsBuildDirectory: "public/build",
```

See #587

File: packages/remix-restify/__tests__/server-test.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+      let request = supertest(createApp());
+      let res = await request.get("/foo/bar");
+
+      expect(res.status).toBe(500);
+      expect(res.text).toMatch(/oh noes/);
+    });
+
+    it("handles status codes", async () => {
+      mockedCreateRequestHandler.mockImplementation(() => async () => {
+        return new Response("", { status: 204 });
```
Comment (MichaelDeBoey) [inline packages/remix-restify/__tests__/server-test.ts:115]:
```suggestion
        return new Response(null, { status: 204 });
```

See #847

File: packages/remix-restify/__tests__/server-test.ts [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        );
+        headers.append(
+          "Set-Cookie",
+          "second=two; MaxAge=1209600; Path=/; HttpOnly; Secure; SameSite=Lax"
+        );
+        headers.append(
+          "Set-Cookie",
+          "third=three; Expires=Wed, 21 Oct 2015 07:28:00 GMT; Path=/; HttpOnly; Secure; SameSite=Lax"
+        );
+        return new Response("", { headers });
```
Comment (MichaelDeBoey) [inline packages/remix-restify/__tests__/server-test.ts:139]:
```suggestion
        return new Response(null, { headers });
```

See #847

**Review by MichaelDeBoey — APPROVED** [review]

File: packages/create-remix/templates/restify/remix.config.js (hunk 1 above)
Comment (MichaelDeBoey) [inline packages/create-remix/templates/restify/remix.config.js:6]:
```suggestion
  assetsBuildDirectory: "public/build",
```

See #587

File: packages/remix-restify/__tests__/server-test.ts (hunk 2 above)
Comment (MichaelDeBoey) [inline packages/remix-restify/__tests__/server-test.ts:115]:
```suggestion
        return new Response(null, { status: 204 });
```

See #847

File: packages/remix-restify/__tests__/server-test.ts (hunk 3 above)
Comment (MichaelDeBoey) [inline packages/remix-restify/__tests__/server-test.ts:139]:
```suggestion
        return new Response(null, { headers });
```

See #847

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Comment by jenseng** [comment]
Thanks @MichaelDeBoey, I made those changes

**Comment by ryanflorence** [comment]
Hey Jon, we're still figuring out the plan for surfacing adapters for all the things without taking on the maintenance burden of having them all in the remix source. I've got an internal note of this PR.

I think the plan is to have a doc that points to all community adapters. You want to publish this yourself somewhere and then we'll make sure it's in the doc when we publish it? (won't be until after the break.)

**Comment by jenseng** [comment]
Sure, works for me 👍

**Pull request closed without merging**